library/cache/src/main/java/com/gtc/getcamp/cache/di/CacheModule.kt

```diff
@@ -0,0 +1,39 @@
+package com.gtc.getcamp.cache.di
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.core.handlers.ReplaceFileCorruptionHandler
+import androidx.datastore.preferences.SharedPreferencesMigration
+import androidx.datastore.preferences.core.PreferenceDataStoreFactory
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.emptyPreferences
+import androidx.datastore.preferences.preferencesDataStoreFile
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import javax.inject.Singleton
+
+private const val USER_PREFERENCES = "user_preferences"
+
+@Module
+@InstallIn(SingletonComponent::class)
+object CacheModule {
+
+    @Singleton
+    @Provides
+    fun providePreferencesDataStore(@ApplicationContext appContext: Context): DataStore<Preferences> {
+        return PreferenceDataStoreFactory.create(
+                corruptionHandler = ReplaceFileCorruptionHandler(
+                        produceNewData = { emptyPreferences() }
+                ),
+                migrations = listOf(SharedPreferencesMigration(appContext,USER_PREFERENCES)),
```

Comment:
why do we need migrations?

Reply:
Şu an için migration' a ihtiyacımız yok. İçeride denerim tüm senaryoları diye eklemiştim. Kaldırıyorum.